pkg/endpoint/endpoint.go

```diff
@@ -166,6 +167,12 @@ func init() {
 	for k, v := range EndpointMutableOptionLibrary {
 		EndpointOptionLibrary[k] = v
 	}
+
+	// This sleep allows detecting k8s.IsEnabled
+	go func() {
+		time.Sleep(2 * time.Second)
```

Comment:
Ouch... please tell me there's other way to do this

Reply:
yeah, I wasn't happy with this. I responded to your comment below. I'll be back :)

Reply:
I moved this to the k8s_watcher. That way the timing is more correct in relation to having k8s setup. It also means that it won't be started in any program that uses the endpoint package (the cilium-health subprogram was running the controller and logging debug)